src/ckb/channel.rs

```diff
@@ -1751,25 +1945,163 @@ impl ChannelActorState {
         AggNonce::sum(nonces)
     }
 
-    pub fn get_all_tlcs(&self) -> Vec<TLC> {
-        debug!("Getting tlcs for commitment tx");
-        debug!(
-            "Current pending offered tlcs: {:?}",
-            self.pending_offered_tlcs
+    pub fn get_active_received_tlcs(&self) -> impl Iterator<Item = &DetailedTLCInfo> {
+        if let ChannelState::ChannelReady(flags) = self.state {
+            if flags.contains(ChannelReadyFlags::AWAITING_REMOTE_REVOKE) {
+                // Remote has not confirmed our latest commitment transaction.
+                // We may have inconsistent view of the active tlcs.
+                // This should not happen.
+                panic!(
+                    "Trying to obtain list of active received tlcs while remote has not confirmed our latest commitment transaction, should have verified we are not in this state before calling this function.",
+                )
+            }
+        }
+        self.tlcs
+            .values()
+            .filter(|detailed_tlc| detailed_tlc.tlc.is_received())
+            .filter_map(|detailed_tlc| match detailed_tlc.removal_info {
+                // Remote has acknowledged the removal of this tlc.
+                Some(_) => None,
+                // This tlc is still active.
+                None => Some(detailed_tlc),
+            })
```

Comment:
suggest to add an `is_removed` fn to DetailedTlcInfo and change the code to:
```suggestion
            .filter(|detailed_tlc| !detailed_tlc.is_removed() && detailed_tlc.tlc.is_received() )
```